.github/workflows/tests-on-demand.yml

```diff
@@ -19,6 +19,10 @@ on:
         type: string
         description: 'Custom event name: In case the `Event name` choice is `custom`, this field is required.'
         required: false
+      wc-version:
+        type: string
+        description: 'If you want to test against a specific WC version, provide it here. Available versions: https://github.com/woocommerce/woocommerce/releases'
```

Comment:
I would add the warning that only some jobs are configured to work with this and most (wp-env) will simply ignore the version.